app/code/community/Ebizmarts/MailChimp/Model/ProcessWebhook.php

```diff
@@ -272,10 +281,53 @@ public function _profile(array $data)
                      */
                     $this->_addSubscriberData($subscriber, $fname, $lname, $email, $listId);
                 }
+
+                if (isset($data['merges']['GROUPINGS'])) {
+                    $this->_processGroupsData($data['merges']['GROUPINGS'], $subscriber, true);
+                }
             }
         }
     }
 
+    protected function _processGroupsData($grouping, $data, $isDataSubscriber = false)
```

Comment:
Change $data variable name, it can be mistaken for the webhook data and it seems to contain a subscriber object.